src/colvarbias_meta.cpp

```diff
@@ -1782,62 +1779,31 @@ int colvarbias_meta::write_replica_state_file()
   if (cvm::debug()) {
     cvm::log("Writing replica state file for bias \""+name+"\"\n");
   }
-  // write down also the restart for the other replicas
-  cvm::backup_file(replica_state_file.c_str());
-  std::ostream *rep_state_os = cvm::proxy->output_stream(replica_state_file);
+
+  std::string const tmp_state_file(replica_state_file+".tmp");
+  std::remove(tmp_state_file.c_str());
+  std::ostream *rep_state_os = cvm::proxy->output_stream(tmp_state_file);
   if (rep_state_os == NULL) {
     cvm::error("Error: in opening file \""+
-               replica_state_file+"\" for writing.\n", FILE_ERROR);
+               tmp_state_file+"\" for writing.\n", FILE_ERROR);
     return FILE_ERROR;
   }
 
   rep_state_os->setf(std::ios::scientific, std::ios::floatfield);
 
   if (!write_state(*rep_state_os)) {
     cvm::error("Error: in writing to file \""+
-               replica_state_file+"\".\n", FILE_ERROR);
-    cvm::proxy->close_output_stream(replica_state_file);
+               tmp_state_file+"\".\n", FILE_ERROR);
+    cvm::proxy->close_output_stream(tmp_state_file);
     return FILE_ERROR;
   }
 
-  cvm::proxy->close_output_stream(replica_state_file);
-
-  // rep_state_os.setf(std::ios::scientific, std::ios::floatfield);
-  // rep_state_os << "\n"
-  //              << "metadynamics {\n"
-  //              << "  configuration {\n"
-  //              << "    name " << this->name << "\n"
-  //              << "    step " << cvm::step_absolute() << "\n";
-  // if (this->comm != single_replica) {
-  //   rep_state_os << "    replicaID " << this->replica_id << "\n";
-  // }
-  // rep_state_os << "  }\n\n";
-  // rep_state_os << "  hills_energy\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy->write_restart(rep_state_os);
-  // rep_state_os << "  hills_energy_gradients\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy_gradients->write_restart(rep_state_os);
-
-  // if ( (!use_grids) || keep_hills ) {
-  //   // write all hills currently in memory
-  //   for (std::list<hill>::const_iterator h = this->hills.begin();
-  //        h != this->hills.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // } else {
-  //   // write just those that are near the grid boundaries
-  //   for (std::list<hill>::const_iterator h = this->hills_off_grid.begin();
-  //        h != this->hills_off_grid.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // }
-  // rep_state_os << "}\n\n";
-  // rep_state_os.close();
+  cvm::proxy->close_output_stream(tmp_state_file);
+
+  if (std::rename(tmp_state_file.c_str(), replica_state_file.c_str())) {
```

Comment:
On Windows only you need to remove before renaming since rename will fail if the target exists, which is the expected case.

Reply:
Basically, this is the same operation as NAMD_backup_file, except adding .tmp to the source rather than .old to the destination.

Reply:
Sounds good.  At any rate, multiple-walkers on Windows?

Reply:
And also an error if source doesn't exist.

Reply:
If you ever want to teach this in a workshop people have Windows laptops, so it would be nice if it didn't just fail and it's easier to fix it now.

Reply:
I realized in the shower this morning that Windows won't let you remove a file that is open by another process. Since that is likely in this case, you will want (on Windows only) to rename the state file to a .old file rather than trying to remove it. It is possible that _unlink() on Windows does not have this limitation. There is a nice discussion of the pain of deleting files on Windows at https://boostgsoc13.github.io/boost.afio/doc/html/afio/FAQ/deleting_open_files.html

Reply:
OK, but then it's almost just as likely that the `.old` file is also still open, especially when files are written in short increments.  Would we need to change the backup extension, à la GROMACS?

Reply:
I'm assuming that the readers will be able finish reading and close the .old file in less time than the update interval. Nothing should be opening the state file after it is renamed to .old.

Reply:
@jcphill The PR is closed so I'm not sure if there will be a notification; the commit for this is cf5d5b2

Reply:
Scratch that, it's ddb312a